packages/react/test/specs/lib/deepmerge-test.ts

```diff
@@ -0,0 +1,118 @@
+import deepmerge from 'src/lib/deepmerge'
+
+describe('deepmerge', () => {
+  test(`always returns an object`, () => {
+    expect(deepmerge({}, {})).toStrictEqual({})
+    expect(deepmerge(null, null)).toStrictEqual({})
+    expect(deepmerge(undefined, undefined)).toStrictEqual({})
+
+    expect(deepmerge(null, undefined)).toStrictEqual({})
+    expect(deepmerge(undefined, null)).toStrictEqual({})
+
+    expect(deepmerge({}, undefined)).toStrictEqual({})
+    expect(deepmerge(undefined, {})).toStrictEqual({})
+
+    expect(deepmerge({}, null)).toStrictEqual({})
+    expect(deepmerge(null, {})).toStrictEqual({})
+  })
+
+  test('undefined overwrites previous value', () => {
+    const merged = deepmerge({ color: 'black' }, { color: undefined })
+    expect(merged).toStrictEqual({ color: undefined })
+  })
+
+  test('null overwrites previous value', () => {
+    const merged = deepmerge({ color: 'black' }, { color: null })
+    expect(merged).toStrictEqual({ color: null })
+  })
+
+  test('undefined gets overwritten  by next value', () => {
+    const merged = deepmerge({ color: undefined }, { color: 'black' })
+    expect(merged).toStrictEqual({ color: 'black' })
+  })
+
```

Comment:
```suggestion
  test('DOES SOMETHING WITH ARRAYS', () => {
    const merged = deepmerge({ color: 'black' }, { color: ['green', 'red'] })
    expect(merged).toStrictEqual({ color: ??? })
  })
```

Can you please a test for this? As we have fallback values that are arrays

Reply:
https://github.com/stardust-ui/react/pull/1907/files#diff-72b4c3a8ba5c9f962bf97db03ae032abR110

Second should always win as I understand. Just want to keep it clear in tests

Reply:
If one of the values is not an object -> last one wins. Tests added, deep merge fixed.